Add a note about aborts encountered while sending early_data

mattcaswell opened the pull request:
In some circumstances it is possible for a client to have a session
reporting a max early data value that is greater than the server will
support. In such cases the client could encounter an aborted connection.

Fixes #6735

richsalz commented on doc/man3/SSL_read_early_data.pod:
```diff
@@ -267,6 +267,19 @@ Nagle's algorithm. If an application opts to disable Nagle's algorithm
+SSL_CTX_set_recv_max_early_data(). Another example is if a server used to
+support TLSv1.3 but was later downgraded to TLSv1.2. Sending early data to such
+a server will cause the connection to abort. Clients that encounter an aborted
+connection while sending early data may choose to retry the connection without
```
s/choose/want/

richsalz commented on doc/man3/SSL_read_early_data.pod:
```diff
@@ -267,6 +267,19 @@ Nagle's algorithm. If an application opts to disable Nagle's algorithm
+support TLSv1.3 but was later downgraded to TLSv1.2. Sending early data to such
+a server will cause the connection to abort. Clients that encounter an aborted
+connection while sending early data may choose to retry the connection without
+sending early data. This does not happen automatically. A client will have to
```
s/. This/, as this/

richsalz commented on doc/man3/SSL_read_early_data.pod:
```diff
@@ -267,6 +267,19 @@ Nagle's algorithm. If an application opts to disable Nagle's algorithm
+connection while sending early data may choose to retry the connection without
+sending early data. This does not happen automatically. A client will have to
+establish a new transport layer connection to the server and attempt the SSL/TLS
+connection again but without sending early data. Note that it is inadvisable to
```
s/is/might be/

mattcaswell replied:
Why only "might be"? Under what conditions would we advise it?

richsalz replied:
Suppose the server got a new revision that included TLS 1.3 support, but some of the new features caused it to crash.

tomato42 replied:
that's a bug in that server and workarounding it should be applied either to none or all handshakes

kaduk replied:
I'm not sure I correctly understand the senes of Rich's discomfort with this text.  FWIW, I would be happy to replace "inadvisable" with "forbidden by the spec".

mattcaswell commented:
Updated commit pushed addressing the feedback so far. I didn't make the last change as I don't think we should be softening that statement. If that's a show stopper, an alternative would be to drop that sentence altogether.

richsalz commented:
Not a show-stopper (I won't -1 it) but I won't approve and am fine if someone else does or if you remove the sentence.

mattcaswell commented:
Ok - fair enough. Ideally I'd prefer to keep the sentence so I'll open it up to @openssl/committers, looking for another reviewer. If no one steps forward, I'll remove it.

mattcaswell commented:
Pushed. Thanks.
